### ISSUE TITLE
Koji integration tests fails on centos stream 9

### ISSUE THREAD

**Issue by teg** [osbuild/osbuild-composer#2322]
See: https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/2112794305

This fails on `main` at the moment.

**Comment by achilleas-k** [comment]
```
Feb 21 06:06:08 ip-10-30-18-43.us-east-1.aws.redhat.com sudo[18236]: ec2-user : PWD=/home/ec2-user/~/builds/osbuild/ci/osbuild-composer ; USER=root ; COMMAND=/bin/tee -a /etc/osbuild-worker/osbuild-worker.toml
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com systemd[1]: Created slice Slice /system/osbuild-worker.
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="Composer configuration:"
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="RelaxTimeoutFactor = 0"
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="base_path = \"\""
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="[koji]"
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="  [koji.localhost]"
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="    [koji.localhost.kerberos]"
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="      principal = \"osbuild-krb@LOCAL\""
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="      keytab = \"/etc/osbuild-worker/client.keytab\""
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="[gcp]"
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="  credentials = \"/etc/osbuild-worker/gcp-credentials.json\""
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="[azure]"
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="  credentials = \"/etc/osbuild-worker/azure-credentials.toml\""
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="[aws]"
Feb 21 06:06:09 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:09Z" level=info msg="  credentials = \"/etc/osbuild-worker/aws-credentials.toml\""
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com sudo[20764]: ec2-user : PWD=/home/ec2-user/~/builds/osbuild/ci/osbuild-composer ; USER=root ; COMMAND=/bin/cp /tmp/osbuild-composer-koji-test/client.keytab /etc/osbuild-worker/client.keytab
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:41Z" level=error msg="Requesting job failed: error requesting job: Post \"http://localhost/api/worker/v1/jobs\": EOF"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:41Z" level=warning msg="Received error from RequestAndRunJob, backing off"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:41Z" level=error msg="Requesting job failed: error requesting job: Post \"http://localhost/api/worker/v1/jobs\": EOF"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[18305]: time="2022-02-21T06:06:41Z" level=warning msg="Received error from RequestAndRunJob, backing off"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com sudo[20788]: ec2-user : PWD=/home/ec2-user/~/builds/osbuild/ci/osbuild-composer ; USER=root ; COMMAND=/bin/systemctl restart osbuild-worker@1
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com systemd[1]: osbuild-worker@1.service: Deactivated successfully.
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="Composer configuration:"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="RelaxTimeoutFactor = 0"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="base_path = \"\""
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="[koji]"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="  [koji.localhost]"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="    [koji.localhost.kerberos]"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="      principal = \"osbuild-krb@LOCAL\""
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="      keytab = \"/etc/osbuild-worker/client.keytab\""
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="[gcp]"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="  credentials = \"/etc/osbuild-worker/gcp-credentials.json\""
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="[azure]"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="  credentials = \"/etc/osbuild-worker/azure-credentials.toml\""
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="[aws]"
Feb 21 06:06:41 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:41Z" level=info msg="  credentials = \"/etc/osbuild-worker/aws-credentials.toml\""
Feb 21 06:06:42 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:42Z" level=info msg="Running job '61038e51-1d2a-43a2-a4d6-8a295849e528' (depsolve)\n"
Feb 21 06:06:42 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:42Z" level=info msg="Running job 'd0187b6b-4665-49b2-b435-c5001a3586c2' (koji-init)\n"
Feb 21 06:06:42 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:42Z" level=info msg="Job 'd0187b6b-4665-49b2-b435-c5001a3586c2' (koji-init) finished"
Feb 21 06:06:53 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:53Z" level=info msg="Job '61038e51-1d2a-43a2-a4d6-8a295849e528' (depsolve) finished"
Feb 21 06:06:53 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:53Z" level=info msg="Running job 'd1e45cce-ded9-437c-a29f-2e573c93345c' (depsolve)\n"
Feb 21 06:06:53 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:53Z" level=info msg="Running job '43fd5dfc-dc21-4362-a506-8acae7e8fbeb' (osbuild-koji)\n"
Feb 21 06:06:53 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:06:53Z" level=info msg="Job 'd1e45cce-ded9-437c-a29f-2e573c93345c' (depsolve) finished"
Feb 21 06:08:06 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:08:06Z" level=error msg="osbuild job failed: osbuild job was unsuccessful" jobId=43fd5dfc-dc21-4362-a506-8acae7e8fbeb
Feb 21 06:08:06 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:08:06Z" level=warning msg="Error reporting job result: error setting job status: 404 — Token not found (IMAGE-BUILDER-WORKER-5)"
Feb 21 06:08:06 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:08:06Z" level=info msg="Job '43fd5dfc-dc21-4362-a506-8acae7e8fbeb' (osbuild-koji) finished"
Feb 21 06:08:06 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:08:06Z" level=info msg="Running job 'efa7b74d-7b45-40f3-8483-8972fccad4ce' (osbuild-koji)\n"
Feb 21 06:18:02 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:18:02Z" level=info msg="osbuild-koji job succeeded" jobId=efa7b74d-7b45-40f3-8483-8972fccad4ce
Feb 21 06:18:02 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:18:02Z" level=warning msg="Error reporting job result: error setting job status: 404 — Token not found (IMAGE-BUILDER-WORKER-5)"
Feb 21 06:18:02 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:18:02Z" level=info msg="Job 'efa7b74d-7b45-40f3-8483-8972fccad4ce' (osbuild-koji) finished"
Feb 21 06:18:02 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:18:02Z" level=info msg="Running job '5ffb6671-30a1-4c7e-b835-0952fc89ba06' (koji-finalize)\n"
Feb 21 06:18:02 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:18:02Z" level=info msg="Job '5ffb6671-30a1-4c7e-b835-0952fc89ba06' (koji-finalize) finished"
```

Looks like the build job succeeds but fails to report.  Very strange that it only happens on CS9.  Gonna keep digging.

**Comment by achilleas-k** [comment]
ah no, missed this
```
level=error msg="osbuild job failed: osbuild job was unsuccessful" jobId=43fd5dfc-dc21-4362-a506-8acae7e8fbeb
```

**Comment by achilleas-k** [comment]
Restarting job to get a manifest

**Comment by achilleas-k** [comment]
odd.  it should be running this https://github.com/osbuild/osbuild-composer/blob/f0f80bd930bb7332718421926210b37fcab978a3/tools/koji-compose-v2.py#L29-L40

**Comment by achilleas-k** [comment]
Where do you see the compose request?  The log is showing me both
```json
{
  "distribution": "centos-9",
  "koji": {
    "server": "https://localhost:4343/kojihub",
    "task_id": 1,
    "name": "name",
    "version": "version",
    "release": "release"
  },
  "image_requests": [
    {
      "architecture": "x86_64",
      "image_type": "guest-image",
      "repositories": [
      ]
    },
    {
      "architecture": "x86_64",
      "image_type": "aws",
      "repositories": [
      ]
    }
  ]
}
```

**Comment by achilleas-k** [comment]
building both manifests locally succeeds so there's nothing obviously wrong there

**Comment by achilleas-k** [comment]
Here are all the jobs files from the most recent run: https://gist.github.com/achilleas-k/f387894f8c802c32a191b5d8088a13ea

**Comment by ondrejbudai** [comment]
`Feb 21 06:08:06 ip-10-30-18-43.us-east-1.aws.redhat.com osbuild-worker[20798]: time="2022-02-21T06:08:06Z" level=warning msg="Error reporting job result: error setting job status: 404 — Token not found (IMAGE-BUILDER-WORKER-5)"`

This is fixed by https://github.com/osbuild/osbuild-composer/pull/2327. Note that this doesn't fix the failing build.

**Comment by ondrejbudai** [comment]
I opened https://github.com/osbuild/osbuild-composer/pull/2329 to investigate.

**Comment by achilleas-k** [comment]
This is likely caused by https://github.com/rpm-software-management/rpm/issues/1833
The issue is fixed in the latest CS9 release of (lib)rpm.  Should be fixed once we get new snapshots: https://github.com/osbuild/rpmrepo/runs/5288368354?check_suite_focus=true

**Comment by teg** [comment]
Are we using the snapshots in all the relevant places?

**Comment by achilleas-k** [comment]
Fixed by #2343
